pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go

```diff
@@ -2372,3 +2373,68 @@ func validateCPUHotplug(field *k8sfield.Path, spec *v1.VirtualMachineInstanceSpe
 	}
 	return causes
 }
+
+func validateVideoConfig(field *k8sfield.Path, spec *v1.VirtualMachineInstanceSpec, config *virtconfig.ClusterConfig) []metav1.StatusCause {
+	var causes []metav1.StatusCause
+
+	if spec.Domain.Devices.Video == nil {
+		return causes
+	}
+
+	if !config.VideoConfigEnabled() {
+		causes = append(causes, metav1.StatusCause{
+			Type:    metav1.CauseTypeFieldValueInvalid,
+			Message: fmt.Sprintf("Video configuration is specified but the %s feature gate is not enabled", featuregate.VideoConfig),
+			Field:   field.Child("video").String(),
+		})
+	}
+
+	if spec.Domain.Devices.AutoattachGraphicsDevice != nil && !*spec.Domain.Devices.AutoattachGraphicsDevice {
+		causes = append(causes, metav1.StatusCause{
+			Type:    metav1.CauseTypeFieldValueInvalid,
+			Message: "Video configuration is not allowed when autoattachGraphicsDevice is set to false",
+			Field:   field.Child("video").String(),
+		})
+	}
+
+	videoDeviceType := spec.Domain.Devices.Video.Type
+	if valid, errMsg := isValidVideoModel(spec.Architecture, videoDeviceType); !valid {
+		causes = append(causes, metav1.StatusCause{
+			Type:    metav1.CauseTypeFieldValueInvalid,
+			Message: errMsg,
+			Field:   field.Child("video").Child("device").String(),
+		})
+	}
+
+	return causes
+}
+
+func isValidVideoModel(architecture, videoDeviceType string) (bool, string) {
+	validVideoModelsByArch := map[string]map[string]struct{}{
+		"amd64": {
+			"vga":    {},
+			"cirrus": {},
+			"virtio": {},
+			"ramfb":  {},
+			"bochs":  {},
+		},
+		"arm64": {
+			"virtio": {},
+			"ramfb":  {},
+		},
+		"s390x": {
+			"virtio": {},
+		},
+		"ppc64le": {
+			"virtio": {},
+			"bochs":  {},
+			"vga":    {},
+			"cirrus": {},
+		},
+	}
+
+	if _, valid := validVideoModelsByArch[architecture][videoDeviceType]; valid {
+		return true, ""
+	}
+	return false, fmt.Sprintf("Video model '%s' is not supported on architecture '%s'", videoDeviceType, architecture)
+}
```

Comment:
The slice version is more compact, although arguably slower... (no change needed, just playing around)
```suggestion
func isValidVideoModel(architecture, videoDeviceType string) (bool, string) {
	validVideoModelsByArch := map[string][]string{
		"amd64":   {"vga", "cirrus", "virtio", "ramfb", "bochs"},
		"arm64":   {"virtio", "ramfb"},
		"s390x":   {"virtio"},
		"ppc64le": {"virtio", "bochs", "vga", "cirrus"},
	}

	if slices.Contains(validVideoModelsByArch[architecture], videoDeviceType) {
		return true, ""
	}
	return false, fmt.Sprintf("Video model '%s' is not supported on architecture '%s'", videoDeviceType, architecture)
}
```

Reply:
nice, cleaner style thanks!